frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx

```diff
@@ -0,0 +1,491 @@
+          <input
+            className="pf-c-form-control"
+            type="text"
+            value={formValues.receiverName}
+            onChange={handleChange}
+            aria-describedby="receiver-name-help"
+            name="receiverName"
+            data-test-id="receiver-name"
+            required
+          />
+        </div>
+        <div className="form-group co-m-pane__dropdown">
+          <label className="control-label co-required">Receiver Type</label>
+          <Dropdown
+            title={'Select Receiver Type...'}
+            name="receiverType"
+            items={receiverTypes}
+            dropDownClassName="dropdown--full-width"
+            data-test-id="receiver-type"
+            selectedKey={formValues.receiverType}
+            onChange={(e) => handleChange({ target: { name: 'receiverType', value: e } })}
+          />
+        </div>
+
+        {formValues.receiverType && (
+          <>
+            <SubForm.Form
+              globals={allGlobals}
+              formValues={formValues}
+              handleChange={handleChange}
+            />
+            <RoutingLabelEditor
+              formValues={formValues}
+              setFormValues={setFormValues}
+              isDefaultReceiver={isDefaultReceiver}
+            />
+          </>
+        )}
+
+        <ButtonBar errorMessage={errorMsg} inProgress={inProgress}>
+          <ActionGroup className="pf-c-form">
+            <Button
+              type="submit"
+              variant="primary"
+              data-test-id="save-changes"
+              isDisabled={isFormInvalid as boolean}
+            >
+              {saveButtonText}
+            </Button>
+            <Button
+              type="button"
+              variant="secondary"
+              data-test-id="cancel"
+              onClick={history.goBack}
+            >
+              Cancel
+            </Button>
+          </ActionGroup>
+        </ButtonBar>
+      </form>
+    </div>
+  );
+};
+
+export const SaveAsDefaultCheckbox = ({ formField, disabled, label, formValues, handleChange }) => {
+  const saveAsDefaultLabelClass = classNames({ 'co-no-bold': disabled });
+  return (
+    <label className={saveAsDefaultLabelClass}>
+      <input
+        type="checkbox"
+        name={formField}
+        data-test-id="save-as-default"
+        onChange={(e) =>
+          handleChange({
+            target: { name: `${formField}`, value: e.target.checked },
+          })
+        }
+        checked={formValues[`${formField}`]}
+        disabled={disabled}
+      />
+      &nbsp; {label}
+    </label>
+  );
+};
+
+const ReceiverWrapper: React.FC<ReceiverFormsWrapperProps> = React.memo(({ obj, ...props }) => {
+  const { alertManagerBaseURL } = window.SERVER_FLAGS;
+  const [alertManagerGlobals, setAlertManagerGlobals] = React.useState();
+  const [inProgress, setInProgress] = React.useState(false);
+  const [errorMsg, setErrorMsg] = React.useState('');
+
+  if (_.isEmpty(errorMsg)) {
```

Comment:
I think there's too much use of `_.isEmpty` and `_.get` in general in these files. It's often unnecessary and it can make it less obvious what the expected data type is.

For example, in this case you could use `const [errorMsg, setErrorMsg] = React.useState()` (default to `undefined`), then you can use `if (!errorMsg)` instead.

I think there are a number of other places in this PR where the lodash funtions could be dropped.

Reply:
Ok, I'll take a pass at reducing these